Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -222,6 +222,12 @@ InjectorFunction dplModelAdaptor(std::vector<OutputSpec> const& filterSpecs, boo
     std::string channelNameForSplitParts;
     static int64_t dplCounter = -1;
     dplCounter++;
+    if (filterSpecs.size() == 1 && DataSpecUtils::match(filterSpecs[0], OutputSpec{{header::gDataOriginAny, header::gDataDescriptionAny}})) {
+      // there should be only a single output channel
+      auto channelName = channelRetriever(filterSpecs[0], 0);
```

Comment:
I think another source of slowdown is the fact that we use the string based sending API. It would be much better to cache the FairMQChannel ptr and use that.